### PR TITLE
refactor: always_specify_types linter rule

### PR DESCRIPTION
**Pull request by mfederowicz**
### Summary

refactor code to pass following linter rules:

- always_specify_types

### CODE REVIEW

**Review by tenshiAMD — APPROVED** [review]

**Comment by mfederowicz** [comment]
@tenshiAMD of course there are some difficult case where I dont know how to specify type, for example:

```
validator: (value) => Validators.requiredField(value),
```

```
final FutureOr Function() onPressed;
```

```
Navigator.of(context).push(
      MaterialPageRoute(
        builder: (BuildContext context) => WorkoutPage(null),
      ),
    );

```

or

```
return defaultValue ?? {};
```

but i dont want to add everywhere :  

```
// ignore: always_specify_types
```
because it is not ok and clean solution :(

**Review by tenshiAMD — CHANGES_REQUESTED** [review]
@mfederowicz Let's wait for @jorre127 to review this one.

**Comment by jorre127** [comment]
@mfederowicz For this one: 

`validator: (value) => Validators.requiredField(value),`

you can just turn it into this: 

`validator: (String? value) => Validators.requiredField(value),`

And then also change the validator in the textField from ` final FormFieldValidator validator;` to   `final FormFieldValidator<String> validator;`

Pretty sure the `final FutureOr Function() onPressed;` can be changed to `final Future<void> Function() onPressed;` seeing as an on pressed shouldn't really return anything, and don't think it does here.

For the navigation we can just specify void as return type. Unless we await the navigation and expect a value to be returned. But most times we don't await so we can just do it like this: 

```
      MaterialPageRoute<void>(
        builder: (BuildContext context) => WorkoutPage(null),
      ),
```

And lastly you can turn `return defaultValue ?? {};` into  ` return defaultValue ?? <String, dynamic>{};` to specify the type of a map.

Can you maybe make these fixes? Then I'll check the PR cause it doesn't seem like all the specify type warnings have been fixed.

**Comment by mfederowicz** [comment]
Ok @jorre127 i fixed code in places you provided, for now there ~~34~~ 25 places to potential fix(improvement) :)

File: lib/src/feature/calendar/widgets/calendar_widget.dart [hunk 1]
```diff
@@ -123,7 +124,8 @@ List<Widget> _getDateColumns() {
   }
 
   for (int i = 0; i < daysInAWeek; i++) {
-    List<Widget> _dates = [];
+    // ignore: always_specify_types
```
Comment (jorre127) [inline lib/src/feature/calendar/widgets/calendar_widget.dart]: Remove this comment

File: lib/src/feature/calendar/widgets/logout_button.dart [hunk 2]
```diff
@@ -10,6 +10,7 @@ class LogoutButton extends StatefulWidget {
   State<LogoutButton> createState() => _LogoutButtonState();
 }
 
+// ignore: always_specify_types
```
Comment (jorre127) [inline lib/src/feature/calendar/widgets/logout_button.dart]: Also remove this one and specify the types in this file

File: lib/src/feature/onboarding/splash/view/splash_screen.dart [hunk 3]
```diff
@@ -54,16 +55,18 @@ class _SplashScreenState extends ConsumerState<SplashScreen> {
   void _navigateToLandingScreen() {
     Navigator.pushReplacement(
       context,
+      // ignore: always_specify_types
```
Comment (jorre127) [inline lib/src/feature/onboarding/splash/view/splash_screen.dart]: Do a search in vscode (or whatever editor you're using) and remove all instances of `// ignore: always_specify_types `then you can easily see what files you have missed.

File: lib/src/feature/authentication/view/landing_screen.dart [hunk 4]
```diff
@@ -61,9 +62,9 @@ class _LandingScreenState extends ConsumerState<LandingScreen> with SingleTicker
 
   @override
   Widget build(BuildContext context) {
-    final _controller = ref.read(AuthenticationController.controller);
+    final AuthenticationController _controller = ref.read(AuthenticationController.controller);
 
-    ref.listen(AuthenticationController.provider, (_, state) {
+    ref.listen(AuthenticationController.provider, (_, Object? state) {
```
Comment (jorre127) [inline lib/src/feature/authentication/view/landing_screen.dart]: Try to be as specific as possible with your types, you should almost never use Object. In this case for example pretty sure this provides a BaseState so change the code to:

` ref.listen<BaseState>(AuthenticationController.provider, (_, BaseState? state) {`

**Comment by jorre127** [comment]
@mfederowicz It's looking good, I think we're almost done. Only it doesn't seem that all the warnings are fixed yet, can you fix these ones? And the ones that pop back up when you delete all the // ignore comments. Feel free to send a message here if you don't know how to specify a type for a specific line

**Comment by mfederowicz** [comment]
@jorre127 almost there, some problematic cases:

```
class _LogoutButtonState extends State<LogoutButton> with Dialogs {
```

```
body: firstLoad
              ? FutureBuilder(
```

```
Expanded(
                flex: 10,
                child: ValueListenableBuilder(
```

```
_notifier = ValueNotifier(widget.totalsWidget.totals);
```

```
Container(
                            child: ValueListenableBuilder(
                              valueListenable: _notifier,
```

**Comment by jorre127** [comment]
@mfederowicz You can change them to:
```
class _LogoutButtonState extends State<LogoutButton> with Dialogs<LogoutButton>
```
```
body: firstLoad
    ? FutureBuilder<List<Workout>>(
```
I used SetData here but you basically just have to match the type here to the type that is passed to the valueListenable property of this widget.
```
Expanded(
  flex: 10,
  child: ValueListenableBuilder<SetData>(
    valueListenable: _notifier,
```
```
_notifier = ValueNotifier<TotalsData>(widget.totalsWidget.totals);
```
and
```
child: ValueListenableBuilder<TotalsData>(
```

**Comment by jorre127** [comment]
@mfederowicz  Can you also maybe from now on put all your changes in seperate commits, because now it's hard to see what are new changes and which ones were already checked.

**Comment by mfederowicz** [comment]
@jorre127 no problemo :P

**Review by jorre127 — APPROVED** [review]

**Comment by mfederowicz** [comment]
ok @tenshiAMD i squashed commits and PR is ready to merge to alpha :)

**Pull request merged**